### PR TITLE
[WIP]Adds in a new traitor item for the Curator called the Shadow Codex

### PR DESCRIPTION
**Pull request by Yokool**
## About The Pull Request

Added in a new traitor item, role restricted for the curator. You can use it in hand, to transfer your mind into a shadow person (not a shadowling), while taking brain damage.

On the first use, the user takes 75 brain damage. On the second and more uses, the user takes 200 brain damage which will kill the user. 

There is a twist to this. You can also target other people. After a short delay, the targeted victim takes the damage instead of you (only ever useful on the second use). Your mind will be transferred to the shadow person as if used in hand. The victim must be living (you can't target a monkey and a few extra rules). 

You can go back into your mind by clicking the book as the shadow person. After a short time, your mind will be transferred back into your original body, leaving the shadowperson as a empty braindead shell (using the book on the shadowperson can lead to fun stuff).

If you die as a shadow person, you suffer additional brain damage (100) and there is also a chance that you will die without going back to your body (i'm pretty sure it should leave you unclonable, quite a risk).

Only you as the curator is able to pick and use the book up, others get paralyzed when picking it up and suffering a little bit of brain damage. They can still pull it.

The item costs 13 TC. Leaving you without an option to buy per say a esword, but you can still buy  a emag. I think its a fair price for such an item.

_This item is heavily inspired by the DS item codex umbra, containing some obvious references to the game._

Note that this is my first **proper** PR and my first "project" I have worked on. Even if you don't like this idea, please leave constructive criticism so I my PRs can be better in the future.

- [ ] Fix the code
- [ ] Make death not a random chance
- [ ] Probably make the shadowminion stronger

## Why It's Good For The Game

The curator is lacking a _major_* unique traitor item. _(*Apart from the magic eight ball which I didn't even know existed before making this)_.

Most of the service roles have a unique traitor item. Like the clown car, mimery guide, his grace, etc. I wanted to add something extra, so people would have something unique to use as a traitor curator. I think it fits the mystical theme of the curator, using books containing ancient texts from lost archives.

This is not supposed to be a kill everyone item while sitting in a locker deep in maintenance (in my view, the book should be used as utility item, since it has some obvious benefits and disbenefits). And because if you die as a shadow person the negatives can leave you with bad traumas only curable by surgery, with a small chance that you will die permanently. Encouraging the player not to take the kill everyone route.

Also the fact that you can use this only on living humans, might encourage the player to capture someone and use their mind, instead of killing them outright on the spot with a boring death.

## Changelog
:cl:
add: Added a new traitor item restricted for the curator, called the shadow codex. Costing 13TC, it can split the users mind, transferring a part of it into a shadow being. It has the ability to target other people, using up their mind instead. With each use, the codex requires more of the users mind, which can lead to instant death.
imageadd: Added the icon for the shadow codex.
/:cl:

### CODE REVIEW

**Comment by Qustinnus** [comment]
insta-kill on a roll is boring, and will just result in me avoiding the item.

File: code/game/objects/items/shadow_codex.dm [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+			shadow_pickup(shadowperson)
+			to_chat(user, "<span class='userdanger'>You return to your body, killing your shadow puppet.</span>")
+		else
+			user.Paralyze(100)
+			to_chat(user, "<span class='userdanger'>You fail to return to your body. Perhaps you should try again.</span>")
+
+/obj/item/shadow_codex/attack_self(mob/user) // when used inhand
+	if(!user.can_read(src))
+		return
+	if(!istype(user, /mob/living/carbon/human)) // might as well leave this here
```
Comment (Qustinnus) [inline code/game/objects/items/shadow_codex.dm]: use the proper helpers for this

**Comment by Yokool** [comment]
> insta-kill on a roll is boring, and will just result in me avoiding the item.

I will do something about it, not sure what yet

**Comment by Shadowflame909** [comment]
Why would you wanna be a shadow person though.

Those fuggers will literally die to everything and anything

**Comment by Yokool** [comment]
> Why would you wanna be a shadow person though.
> 
> Those fuggers will literally die to everything and anything

+They have nightvision
+They have no blood and eyes
+Radimmune, virus immune and they don't need to breathe <-- not that useful
+Heals in darkness

-Shadowpeople take more burn damage (1,5x) 
_welders are a pain but the fact that you see in dark and you should be fighting in dark can help you outrobust an assistant with a welder_
-They slowly die in light

**Comment by Shadowflame909** [comment]
More like rapidly..., and 99.99 the station contains light except for maint.

But y'know whats the one place curator doesn't have access too?

**Comment by Yokool** [comment]
> More like rapidly..., and 99.99 the station contains light except for maint.
> 
> But y'know whats the one place curator doesn't have access too?
> 
> Edit: My point here is, as a pepol who plays a shadow person on Halloween. Your benefit is really a downside. So this item seems likely to only get bought once to see what it is.
> 
> You would literally not be able to roam the halls, otherwise you'll get valided by some shitter shining a light in your face. (Like the chaplain) or the janitor who's replacing lights in your face. Or an evil borg who has his lights on all the time and is maint crawling.
> 
> It's just a VERY hard race to survive with.

Thank you for the long reply. I think I will buff the shadow person to some extent so people have a reason to play it. This is exactly the type of information I needed.

Not sure if taking damage in under lights would defeat the purpose off a shadow person but I think I can come up with something.

**Comment by Shadowflame909** [comment]
Well, as much as I'm hyping up the downsides. Their darkness healing factor should be hyped up as well.

They die SUPER RAPIDLY IN THE LIGHT. Yet, they also HEAL SUPER RAPIDLY IN THE DARKNESS.

They're a pretty good race to shaft mine with, since lavaland is 90% darkness, and constant super speedy healing is really good.

It's just that the opposite happens on the station, which is 90% light.

**Comment by zxaber** [comment]
Imagine the ability to turn other people into shadow people and then couple that with the hypno flash.

An army of darkness.

**Comment by Shadowflame909** [comment]
Who would win, 100 shadow boys.

Or 1 flash light.

The results may surprise you

File: code/game/objects/items/shadow_codex.dm [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	icon_state = "shadowcodex"
+	desc = "A book containing the secrets of shadows, with a large M on the cover. The book is filled with comments from and about the previous owner of this book, which is a mysterious man wearing a dapper suit. You can see <span class=warning>blood</span> on the last page."
+	var/mob/living/carbon/caster = null
+	var/is_reading = 0
+	var/is_converting = 0
+	var/times_used = 0
+	var/mob/living/carbon/human/shadowperson // The variable that holds the summoned minion
+
+/obj/item/shadow_codex/Initialize()
+	to_chat(usr, "<span class=warning>You feel a uncanny presence probing your mind.</span>")
```
[the remark this answers is not in the document]
Comment (Yokool) [reply code/game/objects/items/shadow_codex.dm]: I can't think of a way to not use usr, although I see why it could be bad
sorry for the inconvinience 🆘

**Comment by Yokool** [comment]
@Shadowflame909 
I know we talked about this already but from my tests. It was around 54 seconds while standing in a lit room with a flashlight on me until I was damaged enough making me walk slowly from the damage. And 1 minute and 58 seconds until I was in crit. Idk it doesn't feel that bad but thats up to you.

(I don't want to overexaggerate what I am saying or keep prolonging the debate about this 😃 but please test this on your own)

**Comment by Yokool** [comment]
Actually you know what? I'm going to rework this item into a more interesting state. Probably going to open it again once its finished.

**Comment by Admiral-Gigaton** [comment]
5 years later...

**Pull request closed without merging**